### PR TITLE
1.2

### PR DESCRIPTION
**Pull request by thearchitector**
## Summary

- Reorganizes and improves Noct plugin.
- Adds `SessionedGraphQLRoute` to support authentication and database access from GraphQL queries/mutations.
- Forces dependencies into extras, which are conditionally added during project generation, to reduce the unnecessary overhead for projects that don't use _everything_.
- Ridiculous speedup in unit tests (by mocking dependency installations).
- Update `black` to stable and reformat.

### CODE REVIEW

File: virga/plugins/graphql/route.py [hunk 1]
```diff
@@ -79,12 +69,10 @@ async def _handle_request(self, request: Request) -> Response:
         if result.errors:
             response["errors"] = [format_error(e) for e in result.errors]
 
-        status_code = (
-            status.HTTP_400_BAD_REQUEST if result.errors else status.HTTP_200_OK
+        return ORJSONResponse(
```
Comment (pastap) [inline virga/plugins/graphql/route.py:72]: nice! I forgot graphql never raises errs

Comment (pastap) [reply virga/plugins/graphql/route.py:72]: did you need a test for this? some fake graphql schema that raises an error - to demo the always-200 response and what `errors` will contain?

**Review by pastap — DISMISSED** [review]
looks slick - maybe another test but not overly important

**Comment by thearchitector** [comment]
@pastap alright, because i apparently cant stop myself, i made a few more changes i had intended to make to virga -- need a rereview. nearly exactly the same, only change is how it handles dependencies for generated projects

basically, plugin deps are now in extras (same pattern that fastapi uses), and when you generate a project it adds the extras to the new project. so it means that projects now only require the deps that they actually use/need, instead of all 68 or whatever.

where extras are added: [/files#diff-7f4001978b6bc9047c3943bd5fbc87387f19d22aa3c614e9b1b7aba379ad3041R43-R49=](https://github.com/IndicoDataSolutions/virga/pull/19/files#diff-7f4001978b6bc9047c3943bd5fbc87387f19d22aa3c614e9b1b7aba379ad3041R43-R49=)  
pattern for checking deps: [/files#diff-44b1566f3a2eb10bc8d3257f9322661b67f75e3ff6e3d6abc3d0de3b9f900666R16-R19=](https://github.com/IndicoDataSolutions/virga/pull/19/files#diff-44b1566f3a2eb10bc8d3257f9322661b67f75e3ff6e3d6abc3d0de3b9f900666R16-R19=)

**Review by pastap — APPROVED** [review]
looks fresh! great idea with the import checks

**Pull request merged**